app/src/main/java/com/example/mockprojectv3/repositories/MovieRepository.java

```diff
@@ -1,41 +1,190 @@
+        currentTrending = new ArrayList<>();
+        compositeDisposable = new CompositeDisposable();
+    }
+    public LiveData<List<MovieModel>> getSearchMovies(){
+        return mSearchMovies;
     }
+    public LiveData<List<MovieModel>> getPopularMovies(){
+        return mPopularMovies;
+    }
+    public LiveData<List<MovieModel>> getTrendingMovies(){
+        return mTrendingMovies;
+    }
+    public void loadMoviesApi(int pageNumber) {
+        mPageNumber = pageNumber;
+
+        if (retrieveMovies != null) {
+            retrieveMovies = null;
+        }
 
-    public MovieApiClient getMovieApiClient() {
-        return movieApiClient;
+        retrieveMovies = new RetrieveMovies(pageNumber);
+
+        Single<Response<MovieSearchResponse>> trendingMoviesSingle = retrieveMovies.getTrendingMovies();
+        Single<Response<MovieSearchResponse>> popularMoviesSingle = retrieveMovies.getPopularMovies();
+
+        compositeDisposable.add(
+                Flowable.zip(
+                                popularMoviesSingle.toFlowable(),
+                                trendingMoviesSingle.toFlowable(),
+                                (moviesPopular, moviesTrending) -> {
+                                    List<MovieModel> popularResults = new ArrayList<>();
+                                    List<MovieModel> trendingResults = new ArrayList<>();
+                                    if (moviesPopular.isSuccessful()) {
+                                        popularResults.addAll(moviesPopular.body().getMovies());
+                                    }
+                                    if (moviesTrending.isSuccessful()) {
+                                        trendingResults.addAll(moviesTrending.body().getMovies());
+                                    }
+                                    return new Pair<>(popularResults, trendingResults);
+                                }
+                        )
+                        .subscribeOn(Schedulers.io())
+                        .observeOn(Schedulers.computation())
+                        .subscribe(
+                                pair -> {
+                                    List<MovieModel> popularResults = pair.first;
+                                    currentPopular.addAll(popularResults);
+                                    List<MovieModel> trendingResults = pair.second;
+                                    currentTrending.addAll(trendingResults);
+                                    mPopularMovies.postValue(currentPopular);
+                                    mTrendingMovies.postValue(currentTrending);
+                                },
+                                throwable -> {
+                                    // Xử lý lỗi
+                                    mPopularMovies.postValue(null);
+                                    mTrendingMovies.postValue(null);
+                                }
+                        )
```

Comment:
Handle exception như này:
```
class Resource{};
class Success<T>  extends Resource {
private String message,
private T data;
}
class Loading extends Resource{}
class Error extends Resource{
private String message;
}
```

Khi value của Livedata is Loading ---> Show Loading
is Error ---> hide loading ---> show error
is Success ---> hide loading ---> show data